web/blog/tenzir-v4.16/index.md

```diff
@@ -0,0 +1,83 @@
+---
+title: Tenzir v4.16
+authors: [dominiklohmann]
+date: 2024-06-05
+tags: [release, context, rpm]
+comments: true
+---
+
+Pipelines now connect more flexibly than ever before with [Tenzir
+v4.16](https://github.com/tenzir/tenzir/releases/tag/v4.16.0) and its upgraded
+[`publish`](/operators/publish) and [`subscribe`](/operators/subscribe)
+operators.
+
+![Tenzir v4.16](tenzir-v4.16.excalidraw.svg)
+
+<!-- truncate -->
+
+## Multi-Producer Multi-Consumer
+
+The introduction of the `publish` operator with [Tenzir
+v4.12](/blog/tenzir-v4.12) enabled split-routing of events. We frequently saw
+users write pipelines like this:
+
+```text {0} title="Pipeine 1: Publish alerts"
+…
+| publish alerts
+```
+
+```text {0} title="Pipeine 2: Save alerts with a high risk score to Splunk"
+subscribe alerts
+| where risk_score >= 90
+| fluent-bit splunk …
+```
+
+```text {0} title="Pipeine 3: Save all alerts to S3 for later reference"
+subscribe alerts
+| write json --compact-output
+| compress zstd
+| to s3://bucket/alerts.json.zst
+```
+
+This approach, however, fell apart as soon as another data source tried to
+publish to the topic `alerts`. Trying to do so just displayed an "Address
+already in use" error. Fundamentally, the `publish` and `subscribe` operators
+were single-producer, multi-consumer (SPMC).
+
+With Tenzir v4.16, the `publish` operator's topics no longer have to be unique,
+making it possible to easily merge data flows back together:
+
+```text {0} title="Pipeine 4: Publish further alerts"
+…
+| publish alerts
+```
+
+We think that this seemingly small change makes Tenzir's pipelines more flexible
+than ever before. Now, you can write pipelines for indiviudal parts of your use
+cases.
+
+For example, imagine that you have a pipeline that persists events to "cold
+storage" by writing them to S3 in a strongly compressed format:
+
+```text
+subscribe to-cold-storage
```

Comment:
I think you want something heavy to be there.

Reply:
Please clarify what you mean, I don't get the comment.

Reply:
Ignore, I lost the thought. 😅